common/app/templates/inlineJS/blocking/applyAdfreeRenderCondition.scala.js

```diff
@@ -0,0 +1,133 @@
+@()
+
+(function applyAdfreeRenderCondition(commercial){
+    var AD_FREE_COOKIE = 'gu_adfree_test';
+    var MVT_ID_COOKIE = 'GU_mvt_id';
+    var FIRST_TIME_VISIT_KEY = 'gu.firstVisitTime';
+    var ALREADY_VISITED_KEY = 'gu.alreadyVisited';
+
+    if ((readLocalStorage(ALREADY_VISITED_KEY) || 0) == 0) {
+        writeLocalStorage(FIRST_TIME_VISIT_KEY, new Date().getTime());
+    }
+
+    if (isFirst3VisitDays() && isInVariantGroup()){
+        commercial.showingAdfree = true;
+        return 'is-adfree';
+    } else {
+        commercial.showingAdfree = false;
+        return '';
+    }
+
+    function isFirst3VisitDays() {
+        var firstVisit = readLocalStorage(FIRST_TIME_VISIT_KEY) || 0;
+        return (new Date().getTime() - firstVisit) < 259200000;
+    }
+
+    function isInVariantGroup() {
+        var cookie = readCookie(AD_FREE_COOKIE);
+        if (cookie == null) {
+            assignUser();
+        }
+        return readCookie(AD_FREE_COOKIE) == "variant";
+    }
+
+    function isIOS() {
+        return /(iPad|iPhone|iPod touch)/i.test(navigator.userAgent);
+    }
+
+    function isSafari() {
+        return /(safari)/i.test(navigator.userAgent);
+    }
+
+    function assignUser() {
+        var mvtNumValues = 899999;
+        var audience = 0.1; // keep this in sync with new-user-adverts-disabled.js
+        var audienceOffset = 0; // keep this in sync with new-user-adverts-disabled.js
+        var smallestTestId = mvtNumValues * audienceOffset;
+        var largestTestId  = smallestTestId + mvtNumValues * audience;
+        var mvtCookieId = readCookie(MVT_ID_COOKIE);
+        var variants;
+        var testVariantIndex;
+
+        var alreadyVisited = readLocalStorage('gu.alreadyVisited') || 0;
+        if (alreadyVisited == 0
+            && mvtCookieId
+            && mvtCookieId > smallestTestId
+            && mvtCookieId <= largestTestId
+            && !isIOS() && !isSafari()) {
+            variants = [{
+                id: 'sampletest-variant',
+                test: function () {
+                    writeCookie(AD_FREE_COOKIE, 'sampletest-variant');
+                }
+            }, {
+                id: 'sampletest-control',
+                test: function () {
+                    writeCookie(AD_FREE_COOKIE, 'sampletest-control');
+                }
+            }];
+            testVariantIndex = mvtCookieId % variants.length;
+            variants[testVariantIndex].test();
+        } else {
+            writeCookie(AD_FREE_COOKIE, 'notintest');
+        }
+    }
+
+    function readLocalStorage(key) {
+        if ("localStorage" in window) {
+            var data;
+            var dataParsed;
+            data = localStorage.getItem(key);
```

Comment:
I know this is already merged, but `"localStorage" in window` is not enough to guarantee that calls to `setItem` and `getItem` work without throwing. See for instance #8102 
If you can't use the utility storage I'd advise to copy paste the isStorageAvailable method or wrap everything in try / catch